### DISCUSSION TITLE
How can I use Open-webui as the frontend for my app?

### DISCUSSION THREAD

**Discussion by rcontesti** [open-webui/open-webui#6800] [Q&A]
I have a quite complex app that uses several LLM models and also calls databases. Given an input, produces and output. The app is available through API POST and requests. I wish I could use open-webui as a frontend for my app but I'm not sure how. Thank you!

**Comment by reecelikesramen** [comment]
Something has to be done about the way the server handles origins right now. Because credentials get sent, you need to set explicit origins since wildcards are forbidden with that option. If you have a reverse proxy like nginx you can get around this by just setting the origin to the request origin per-request, but *this isn't a good idea for production code*. Once you've done this you just have to modify `WEBUI_HOST` in `constants.ts` and serve your frontend.
